pkg/secrets/scrub.go

```diff
@@ -12,12 +12,19 @@ const (
 	scrubStructTag = "scrub"
 	// scrubTagAlways is a scrub tag type used to indicate a field is a credential
 	scrubTagAlways = "always"
+	// scrubTagMapValues is a scrub tag type used to indicate a map[string]string field that contain keys with secrets.
+	// List of scrubbed keys are listed in mapKeysToReplace.
+	scrubTagMapValues = "map-values"
 	// scrubTagDependent is a scrub tag type used to indicate a field is dependent on credentials and could be used to exfiltrate credentials
 	scrubTagDependent = "dependent"
 	// ScrubReplacementStr is a string format of a masked credential
 	ScrubReplacementStr = "******"
 )
 
+var (
+	mapKeysToReplace = []string{"client_secret"}
```

Comment:
Potentially we could add another tag that would specify map keys to replace within proto, something like `scrub-map-keys`. I'll leave it to you to decide whether to pursue that or not

Reply:
That is a good idea. I'll keep the implementation as it is. But if we come back to this functionality we could consider adding that.